src/render-if-visible.tsx

```diff
@@ -56,15 +60,19 @@ const RenderIfVisible = ({
     }
   }, [isVisible, intersectionRef])
 
-  return (
-    <div ref={intersectionRef}>
-      {isVisible ? (
-        <>{children}</>
-      ) : (
-        <div style={{ height: placeholderHeight.current }} />
-      )}
-    </div>
-  )
+  return React.createElement(
+        rootElement,
+        {
+            children: isVisible
+                ? (<>{children}</>)
+                : React.createElement(
+                    placeholderElement,
+                    { className: "renderIfVisible-placeholder", style: placeholderStyle }
```

Comment:
Where does `placeholderStyle` come from?

Reply:
Sorry you are correct, I moved the {height: placeholderHeight.current}  into a const locally but didn't do that when migrating to this edit. Let me try adding that. (sorry new to github world)